OpenShift Pipelines Release Notes 1.17 (#85590)
openshift/openshift-docs: modules/op-release-notes-1-17.adoc

// This module is included in the following assemblies:
// * release_notes/op-release-notes-1-16.adoc

:_mod-docs-content-type: REFERENCE
[id="op-release-notes-1-17_{context}"]
= Release notes for {pipelines-title} General Availability 1.17

With this update, {pipelines-title} General Availability (GA) 1.17 is available on {OCP} 4.15 and later versions.

[id="new-features-1-17_{context}"]
== New features

In addition to fixes and stability improvements, the following sections highlight what is new in {pipelines-title} 1.17:

[id="pipelines-new-features-1-17_{context}"]
=== Pipelines

> * With this release, you can configure multiple Git providers, or you can add multiple configurations for the same Git provider, in the `TektonConfig` custom resource (CR). You can then specify a Git configuration to use in different task runs and pipeline runs by using the `configKey` parameter.

Reviewer: I am not sure how to phrase that but maybe we need to clarify this is a feature for tekton resolvers and not about Pipelines-as-Code

Author: Good point, I am changing it to `With this release, you can configure multiple Git providers by using the Git resolver,...` I think that would work better, WDYT?

 	  params:
  	# ...
  	- name: configKey
   	  value: test1
# ...
----

* With this update, you can now monitor the running `PipelineRun` resources at the namespace, pipeline, and pipeline run level, in addition to the cluster level. You can configure the level through the `TektonConfig` custom resource by setting the `metrics.running-pipelinerun.level` parameter value to `namespace`, `pipeline`, or `pipelinerun`. The default value is empty, which sets the monitoring at cluster level.
+
.Example of monitoring the running pipeline run at the namespace level
[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  pipeline:
    # ...
    metrics.running-pipelinerun.level: namespace
# ...
----

[id="cli-new-features-1-17_{context}"]
=== CLI

* With this update, you can run the `skopeo-copy` command with additional arguments, for example, `--all` and `--preserve-digests`. You can pass these arguments as a space-separated string, as shown in the following example:
+
.Example command
[source,terminal]
----
$ tkn task start skopeo-copy ARGS=”--all --preserve-digests”
----

[id="pac-new-features-1-17_{context}"]
=== {pac}

> * With this update, the `pipelines_as_code_running_pipelineruns_count` metric is added to show the number of running `PipelineRun` resources in {pac}.
> 

Reviewer 2: ```suggestion
* With this update, the `pipelines_as_code_running_pipelineruns_count` metric is added to show the number of running `PipelineRun` resources in {pac}. The metric can be filtered by repository or by namespace.

```

> * With this update, the `pipelines_as_code_pipelinerun_duration_seconds_sum` metric is added to show the total duration in seconds taken by all `PipelineRun` resources in {pac}. The metric can be filtered by repository or by namespace.

Reviewer: cc @zakisk

[id="tekton-chains-new-features-1-17_{context}"]
=== {tekton-chains}

* With this update, you can configure the `TektonConfig` custom resource (CR) to generate the `x509` key pair of `ecdsa` type and use it with {tekton-chains} to sign artifacts. You can generate the key pair by setting the the `generateSigningSecret` field in the `TektonConfig` custom resource (CR) to `true`:
+
.Example of creating ecdsa key pair
[source,yaml]
----
apiVersion: operator.tekton.dev/v1
kind: TektonConfig
metadata:
  name: config
spec:
# ...
  chain:
    disabled: false
    generateSigningSecret: true 
# ...
----

> * With this update, the default values for `artifacts.taskrun.storage` and `artifacts.pipelinerun.storage` {tekton-chains} parameters are changed from `tekton` to `oci` to enhance the initial setup and performance of {tekton-chains}.

Reviewer 3: The values are not changed, if user has no default values, then default values are set, and also we are setting default values for `artifacts.oci.storage`

cc @jkhelil can you please elaborate a bit more here

Author: Wait I am confused. We have one whole (already merged) issue about this change. From `tekton` to `oci`:. It will be reflected in the docs as follows: https://84253--ocpdocs-pr.netlify.app/openshift-pipelines/latest/secure/using-tekton-chains-for-openshift-pipelines-supply-chain-security.html#supported-parameters-tekton-chains-configuration_using-tekton-chains-for-openshift-pipelines-supply-chain-security

Author: The `artifacts.oci.storage` is already set to `oci` by default according to our docs, so no change there: https://docs.openshift.com/pipelines/1.16/secure/using-tekton-chains-for-openshift-pipelines-supply-chain-security.html#chains-supported-parameters-oci_using-tekton-chains-for-openshift-pipelines-supply-chain-security Am I missing something?

* With this update, {tekton-chains} now supports extracting the `mongo-server-url` URL from a specified file that can have any name. You can now use the `storage.docdb.mongo-server-url-path` parameter pointing to a valid file path within the container.

[id="breaking-changes-1-17_{context}"]
== Breaking changes

* With this update, the deprecated `ClusterTask` resource is removed from the Operator. As an alternative, you can use the cluster resolver.

* With this update, the community cluster tasks are removed from the Operator. As an alternative, you can download them from the link:https://github.com/tektoncd/catalog[Tekton catalog] (GitHub resource). The community cluster tasks are planned to be added as tasks in the future release. The following list shows the removed community cluster tasks:

** `argocd-task-sync-and-wait`
** `git-cli`
** `helm-upgrade-from-repo`
** `helm-upgrade-from-source`
** `jib-maven`
** `kubeconfig-creator`
** `pull-request`
** `trigger-jenkins-job`

[id="fixed-issues-1-17_{context}"]
== Fixed issues

> * With this update, the {pac} controller no longer processes the GitLab push event if the push event contains no commit. Instead, it correctly displays an error message warning the user that no commit is attached.
> 

Reviewer 2: ```suggestion
* With this update, the {pac} controller no longer processes the GitLab push event if the push event payload contains no commit. Instead, it correctly displays an error message warning the user that no commit is attached.

```

// This module is included in the following assemblies:

> // * release_notes/op-release-notes-1-16.adoc

Reviewer 4: ```suggestion
// * release_notes/op-release-notes-1-17.adoc
```

      test2.server-url: "api.internal-gitlab.com"
      test2.api-token-secret-name: "test2-secret"
      test2.api-token-secret-key: "pat"
      test2.api-token-secret-namespace: "test2"
      test2.default-org: "tektoncd-infra"
# ...
----
+
.Example of using a `test1` configuration in a pipeline run
[source,yaml]
----
apiVersion: tekton.dev/v1
kind: PipelineRun
metadata:
  name: example
spec:
  pipelineRef:
    resolver: git
    params:
    # ...
    - name: configKey
      value: test1
# ...
----

* With this update, you can now monitor the running `PipelineRun` resources at the namespace, pipeline, and pipeline run level, in addition to the cluster level. You can configure the level through the `TektonConfig` custom resource by setting the `metrics.running-pipelinerun.level` parameter value to `namespace`, `pipeline`, or `pipelinerun`. The default value is empty, which sets the monitoring at cluster level.
+
.Example of monitoring the running pipeline run at the namespace level
[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  pipeline:
    # ...
    metrics.running-pipelinerun.level: namespace
# ...
----

> [id="Operator-new-features-1-17_{context}"]

Reviewer 4: ```suggestion
[id="operator-new-features-1-17_{context}"]
```

> === CLI

Reviewer 5: This has nothing to do with CLI, it's about ecosystem tasks, maybe section "Operator"?

> * With this update, you can run the `skopeo-copy` command with additional arguments, for example, `--all` and `--preserve-digests`. You can pass these arguments as a space-separated string, as shown in the following example:

Reviewer 6: This seems to belong in CLI not Operator?

+
.Example command
[source,terminal]
----

> $ tkn task start skopeo-copy ARGS=”--all --preserve-digests”

Reviewer 5: ```suggestion
$ tkn task start skopeo-copy -p ARGS="--all --preserve-digests"
```

----

[id="pac-new-features-1-17_{context}"]
=== {pac}

* With this update, the `pipelines_as_code_running_pipelineruns_count` metric is added to show the number of running `PipelineRun` resources in {pac}. The metric can be filtered by repository or by namespace.

> * With this update, the `pipelines_as_code_pipelinerun_duration_seconds_sum` metric is added to show the total duration in seconds taken by all `PipelineRun` resources in {pac}. The metric can be filtered by repository or by namespace.

Reviewer 2: ```suggestion
* With this update, the `pipelines_as_code_pipelinerun_duration_seconds_sum` metric is added to show the total duration in seconds taken by all `PipelineRun` resources in {pac}. The metric can be filtered by repository, namespace, `PipelineRun` status, and `PipelineRun` status change reason.
```

Author: Hello @adellape I am just pinging you so that you are aware of this one late change coming to the PR, rest is the same, thank you!

[id="tekton-chains-new-features-1-17_{context}"]
=== {tekton-chains}

> * With this update, you can configure the `TektonConfig` custom resource (CR) to generate the `x509` key pair of `ecdsa` type and use it with {tekton-chains} to sign artifacts. You can generate the key pair by setting the the `generateSigningSecret` field in the `TektonConfig` custom resource (CR) to `true`:

Reviewer 6: of the `ecdsa` type

+

> .Example of creating ecdsa key pair

Reviewer 4: ```suggestion
.Example of creating `ecdsa` key pair
```

> .Example of creating `ecdsa` key pair

Reviewer 6: creating an `ecdsa` key pair

[source,yaml]
----
apiVersion: operator.tekton.dev/v1
kind: TektonConfig
metadata:
  name: config
spec:
# ...
  chain:
    disabled: false
    generateSigningSecret: true 
# ...
----

* Before this update, if you did not configure {tekton-chains} in the `TektonConfig` CR, the Operator would not pass any default Chains configuration.
With this update, if you do not configure {tekton-chains} in the `TektonConfig` CR, the Operator sets the {tekton-chains} configuration with these default properties:
+
[source,yaml]
----
apiVersion: operator.tekton.dev/v1
kind: TektonConfig
metadata:
  name: config
spec:
# ...
  chain:
    artifacts.taskrun.format: in-toto
    artifacts.taskrun.storage: oci
    artifacts.oci.storage: oci
    artifacts.oci.format: simplesigning
    artifacts.pipelinerun.format: in-toto
    artifacts.pipelinerun.storage: oci
# ...
----

* With this update, {tekton-chains} now supports extracting the `mongo-server-url` URL from a specified file that can have any name. You can now use the `storage.docdb.mongo-server-url-path` parameter pointing to a valid file path within the container.

[id="breaking-changes-1-17_{context}"]
== Breaking changes

> 

Reviewer 6: As an alternative, you can use the cluster revolver to access tasks that {pipelines-shortname} installs in the `openshift-pipelines` namespace.

* With this update, the deprecated `ClusterTask` resource is removed from the Operator. As an alternative, you can use the cluster resolver.

> * With this update, the community cluster tasks are removed from the Operator. As an alternative, you can download them from the link:https://github.com/tektoncd/catalog[Tekton catalog] (GitHub resource). The community cluster tasks are planned to be added as tasks in the future release. The following list shows the removed community cluster tasks:

Reviewer 4: ```suggestion
* With this update, the community cluster tasks are removed from the Operator. As an alternative, you can download them from the link:https://github.com/tektoncd/catalog[Tekton catalog] (GitHub resource). The community cluster tasks are planned to be added as tasks in a future release. The following list shows the removed community cluster tasks:
```
